tests/test_mask_creation.py

```diff
@@ -51,3 +50,28 @@ def test_create_sidecar_mask(data):
     # Check the .msk file, too.
     with rasterio.open(str(data.join('RGB.byte.tif.msk'))) as msk:
         assert (mask == msk.read(1, masked=False)).all()
+
+
+def test_create_mask_windowed_sidecar(data):
+    """Writing masks by window succeeds with sidecar mask
+    """
+    with rasterio.Env(GDAL_TIFF_INTERNAL_MASK=False):
+        with rasterio.open(str(data.join('RGB.byte.tif')), 'r+') as dst:
```

Comment:
The test data is copied to temp directory using the `data` fixture, [defined in conftest.py](https://github.com/mapbox/rasterio/blob/master/tests/conftest.py#L42). Not immediately obvious I admit, but it was keeping with the pattern of the existing tests in that file.